daemon/info.go

```diff
@@ -286,38 +287,37 @@ func (daemon *Daemon) fillDefaultAddressPools(ctx context.Context, v *system.Inf
 func hostName(ctx context.Context) string {
 	ctx, span := tracing.StartSpan(ctx, "hostName")
 	defer span.End()
-	hostname := ""
-	if hn, err := os.Hostname(); err != nil {
-		log.G(ctx).Warnf("Could not get hostname: %v", err)
-	} else {
-		hostname = hn
+	hn, err := os.Hostname()
+	if err != nil {
+		log.G(ctx).WithError(err).Warn("Could not get hostname")
+		return ""
 	}
-	return hostname
+	return hn
 }
 
 func kernelVersion(ctx context.Context) string {
 	ctx, span := tracing.StartSpan(ctx, "kernelVersion")
 	defer span.End()
 
-	var kernelVersion string
+	var ver string
 	if kv, err := kernel.GetKernelVersion(); err != nil {
-		log.G(ctx).Warnf("Could not get kernel version: %v", err)
+		log.G(ctx).WithError(err).Warn("Could not get kernel version")
 	} else {
-		kernelVersion = kv.String()
+		ver = kv.String()
 	}
-	return kernelVersion
+	return ver
 }
 
 func memInfo(ctx context.Context) *meminfo.Memory {
 	ctx, span := tracing.StartSpan(ctx, "memInfo")
 	defer span.End()
 
-	memInfo, err := meminfo.Read()
+	mi, err := meminfo.Read()
 	if err != nil {
 		log.G(ctx).Errorf("Could not read system memory info: %v", err)
```

Comment:
Oh! Looks like I missed  one;  let  me update this one as well while  I'm at it 🙈